web/app.js

```diff
@@ -1314,7 +1314,10 @@ const PDFViewerApplication = {
           PDFJSDev.test("FIREFOX || MOZCENTRAL")
         ) {
           // Telemetry labels must be C++ variable friendly.
-          const versionId = `v${info.PDFFormatVersion.replace(".", "_")}`;
+          let versionId = "other";
+          if (info.PDFFormatVersion) {
```

Comment:
Strictly speaking, we should probably verify that the version is one of the accepted values before assigning to the variable. The PDFFormatVersion field can have any value, as seen at https://github.com/mozilla/pdf.js/blob/81af207428b2fafdf4373ad6d2689928f72ee2d6/src/core/document.js#L513
and https://github.com/mozilla/pdf.js/blob/81af207428b2fafdf4373ad6d2689928f72ee2d6/src/core/document.js#L632

Reply:
Fair point, I've added validation of the same format as for the generator just below; thanks for the review!

Reply:
The existing format looks needlessly complicated. Let's change it to:

```
if (KNOWN_VERSIONS.includes(info.PDFFormatVersion)) {
  versionId = `...`;
}
```

Reply:
Fixed; now there's less copy-and-paste in the patch :-)